jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.configurations.CommandLineState
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.process.ProcessHandlerFactory
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.util.io.FileUtil
+import software.aws.toolkits.jetbrains.settings.SamSettings
+
+class SamRunningState(
+    environment: ExecutionEnvironment,
+    val settings: SamRunSettings
+) : CommandLineState(environment) {
+    lateinit var codeLocation: String
+
+    override fun startProcess(): ProcessHandler {
+        val samCliExecutable = SamSettings.getInstance().executablePath
+        val commandLine = GeneralCommandLine(samCliExecutable)
+            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.NONE)
+            .withParameters("local")
+            .withParameters("invoke")
+            .withParameters("--template")
+            .withParameters(createSamTemplate())
+            .withParameters("--event")
+            .withParameters(createEventFile())
+            .withEnvironment(settings.environmentVariables)
+
+        return ProcessHandlerFactory.getInstance().createColoredProcessHandler(commandLine)
+    }
+
+    private fun createSamTemplate(): String {
```

Comment:
love it! ultimately obviously we'll want to detect this in the project and only fall back to the 'JIT' template if necessary.

Reply:
Agree will cut another issue to track using template

Reply:
Created #261